cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -23,6 +23,8 @@ namespace cudf::io::detail::parquet {
 
 namespace {
 
+auto constexpr max_decimal128_precision = 38;
```

Comment:
Perhaps comment where this number comes from?

$$
\lfloor \log_{10}{(2^{8 n - 1} - 1)} \rfloor
$$

with $n = 16$

Aside; for all $n \ge 1$ this approximation

$$
\lfloor (8 n - 1) \log_{10}{2} \rfloor
$$

is actually exact.

So one could (although arguably it's not really any clearer) use:
```c++
  auto constexpr max_supported_decimal_width = (sizeof(__int128_t) * CHAR_BIT) / 8;
  /* log10(2) = 0.3010299956639812 */
  std::uint32_t constexpr max_decimal_precision = (8 * max_supported_decimal_width - 1) * 0.3010299956639812;
```

Reply:
Added